### DISCUSSION TITLE
Standard Notes

### DISCUSSION THREAD

**Discussion by bagelwaffle** [catppuccin/catppuccin#2097] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the suggested port be themed?

through CSS

options are listed in the [official documention](https://standardnotes.com/help/plugins/themes)

### Any additional comments?

The theme(s) can be added through Standard Notes' builtin plugin system
More information can be found at https://github.com/standardnotes/plugins/tree/main

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by sarchotic** [comment]
There hasn't been any updates on this, has anyone started working on it yet?

**Reply by ghost** [reply]
I'm going to start reading the Catppuccin docs and see if I'm able to help with this. I'll work on Latte if no one else is.

**Reply by ghost** [reply]
Actually, I've tried to make a Standard Notes account and kept getting errors. The desktop app is also super bloated and broken: if you get served a captcha then you can't actually complete it. I also don't like that my important but not private notes are encrypted, so I'm not actually going to work on Standard Notes.

**Reply by sarchotic** [reply]
Ok, are you still going to work on Latte? I will go ahead and start working on one myself if you won't be working on it.

**Reply by ghost** [reply]
I can't make an account due to their backend. I can't test it.

What I can offer is use this project as practice for cross-referencing documentation and provide a hypothetical StandardNotes theme that has no warranty. I am using this issue as a good first issue.

I don't think StandardNotes has an SDK that provides automated tests.

**Reply by sarchotic** [reply]
I could start working on it as well, could collab

**Comment by onexbash** [comment]
Oh my god this would be so cool to have a catppuccin theme for Standard Notes. Its by far my favorite note taking app because its so minimal.

**Comment by ghost** [comment]
As much as I'd like to see Catppuccin in Standard Notes, I don't think I can do it at my skill level. Also, it turns out I'm just not interested in tools like Yarn and Node, and I thought that the docs would be more than a couple examples with no build instructions.

I also can't figure out Standard Notes for the life of me. I try to add the plugins from their GitHub repo but it says there's an error with the URL. I also can't make an account for some reason.

The repository is still at jacko-png/standard-notes-catppuccin-latte, but I haven't done much. Just the first few colours, because I don't know what ‘foreground’ should be.

Sorry. I don't know what else to say.

**Reply by sarchotic** [reply]
No issue, I will go ahead start working on a version for mocha and everythinng 👍

**Reply by uncenter** [reply]
If you haven't already, you should look into using https://github.com/catppuccin/whiskers to generate all flavors from one template!

**Reply by sarchotic** [reply]
👍 Thanks for the pointer, I'll check it out!

**Comment by JoeC-Dev** [comment]
Anyone else working on this?

**Comment by JoeC-Dev** [comment]
Okay I think I figured it out, at least for mocha. I think they all need to be separate repos though???? Can anyone give me guidelines for how to submit them to be added to the ports??? I need to just fork my mocha repo and make one for the other 3 colorways but that should be quick

**Reply by bagelwaffle** [reply]
Hello! All of them should be in the same repo. The [style guide](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md) and the [port creating guide](https://github.com/catppuccin/catppuccin/blob/main/docs/port-creation.md) have all the info you need to get started!

Also it's recommended to use [whiskers](https://whiskers.catppuccin.com/getting-started/introduction/) to generate the flavours from a template.

Let me know if you need help with anything (except whiskers idk how it works)!

**Reply by JoeC-Dev** [reply]
Hmm. I wonder if thats possible??? SN requires the themes to be in github releases. Along with a package file that I dont think you specify path to. It looks for it at the top level of the repo. That file Needs to differ per theme.

**Reply by bagelwaffle** [reply]
Oh I didn't realize it's like that. I think the way to go about it then is to have four different branches in addition to main and seperate it that way, but you might want to check with the people in the catppuccin discord.

Re: GitHub releases
From what I see in the [dracula port](https://github.com/dracula/standard-notes/blob/master/ext.json), you should be able to point the download_url to any zip file you put in gh releases.

**Reply by JoeC-Dev** [reply]
Ill take another look later. Multiple branches seems like a good idea! Based on the linking and stuff it seems like it should work

**Reply by JoeC-Dev** [reply]
Oh also one other question I have is how does Catppuccin handle things that need to be hosted?? Part of SN plugins requires a hosted file that is reachable from SN. It's a json that gives info on the git repo and has links on where the application can pull the theme / plugin. RN I am following the guidance of SN and I am hosting it with there proprietary blog. This hinges on my account. Its a private page linked to my blog (which I only use for this lol). Everyone else does this too with their plugins BUT I dont know if catppuccin wants the theme to specifically be linked to my account? Again you cant see anything about my account, it just relys on me to keep mine forever.

**Reply by bagelwaffle** [reply]
You should be able to put the JSON file on github and use the "raw" link in SN.

**Reply by JoeC-Dev** [reply]
Alright. Ill work on a mono-repo for all this when i have a chance. I think the JSON might need to be hosted so a github io should work. I dont think SN likes raw. I tried it before and for some reason it didnt work.

**Reply by bagelwaffle** [reply]
That's strange! You could ask if they would host on it on `catppuccin.com` when you send it for review or I suppose you could just use a cdn jsdelivr link pointing to the file on github ([example](https://draculatheme.com/standard-notes)).
